lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -17,11 +18,13 @@ def initialize(engine_config:)
 
         def run(file)
           process_file(file)
-        rescue *RESCUABLE_ERRORS => ex
-          $stderr.puts("Skipping file #{file} due to exception (#{ex.class}): #{ex.message}\n#{ex.backtrace.join("\n")}")
         rescue => ex
-          $stderr.puts("#{ex.class} error occurred processing file #{file}: aborting.")
-          raise ex
+          if RESCUABLE_ERRORS.map { |klass| ex.instance_of?(klass) }.include?(true)
```

Comment:
Before anybody asks: `ActiveSupport` is not included in this gem, and it doesn't seem worth adding just for this.